src/main/scala/OiOsmDiff.scala

```diff
@@ -0,0 +1,91 @@
+package oiosmdiff
+
+import java.net.URI
+
+import com.typesafe.scalalogging.LazyLogging
+
+import geotrellis.proj4.{WebMercator, LatLng}
+import geotrellis.spark.SpatialKey
+import geotrellis.spark.tiling.ZoomedLayoutScheme
+import geotrellis.vector.{Geometry, Feature}
+import geotrellis.vectortile._
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.functions._
+import org.apache.spark.rdd.RDD
+
+import org.locationtech.jts.geom.{Geometry => JTSGeometry}
+
+import vectorpipe._
+import vectorpipe.GenerateVT.VTF
+import vectorpipe.functions.osm._
+
+class OiOsmDiff(
+    osmOrcUri: URI,
+    oiGeoJsonUri: URI
+)(@transient implicit val ss: SparkSession)
+    extends LazyLogging
+    with Serializable {
+  def saveTilesForZoom(zoom: Int, outputS3Prefix: URI): Unit = {
+    val badRoads = Seq("proposed", "construction", "elevator")
+
+    val osmData = OSM.toGeometry(ss.read.orc(osmOrcUri.toString))
+    val isValid: (JTSGeometry) => Boolean = (jtsGeom: JTSGeometry) => jtsGeom.isValid()
+    val isValidUDF = udf(isValid)
+
+    val osmRoadData = osmData
+      .select("id", "_type", "geom", "tags")
+      .withColumn("roadType", osmData("tags").getField("highway"))
+      .withColumn("surfaceType", osmData("tags").getField("surface"))
+
+    val osmRoads = osmRoadData
+      .filter(isRoad(col("tags")))
+      .where(
+        (osmRoadData("geom").isNotNull && isValidUDF(osmRoadData("geom"))) &&
```

Comment:
We found that filtering on `isValidUDF` in a separate `where` helps improve performance.

Reply:
Updated. Didn't seem to make much of a difference, perhaps a few seconds? But this is a much smaller job.

What type of performance gains were you seeing on the jobs you were running?